pshtt/pshtt.py

```diff
@@ -775,18 +828,31 @@ def is_domain_enforces_https(domain):
 
 
 def is_domain_strong_hsts(domain):
-    return (
-        is_hsts(domain) and
-        hsts_max_age(domain) >= 31536000
-    )
+    if is_hsts(domain) and hsts_max_age(domain):
+        return (
+            is_hsts(domain) and
+            hsts_max_age(domain) >= 31536000
+        )
+    return None
+
+
+# Checks if the domain had an Unknown error somewhere
+# The main purpos of this is to flag any odd websites for
+# further debugging with other tools.
+def did_domain_error(domain):
+    list_of_endpoints = domain.to_object()
+    endpoints = ['http', 'https', 'httpwww', 'httpswww']
+    for endpoint in endpoints:
+        if list_of_endpoints[endpoint]['unknown_error']:
+            return True
```

Comment:
I'd prefer to follow the pattern of other methods and explicitly do a big `or` statement between the 4 endpoints. Though less efficient, I think that makes the intended logic more readable at a glance.